src/useValidation/ValidationExecutor.ts

```diff
@@ -18,23 +18,23 @@ export class ValidationRuleExecutor<T> {
     this.controller = new AbortController()
   }
 
-  public async validate({ value, name, rules, source, previousValue }: Validate<T>): Promise<string | void> {
+  public async validate({ value, name, rules, source, previousValue }: Validate<T>): Promise<string> {
     const { signal } = this.controller
 
     for (const rule of rules) {
-      if (signal.aborted) {
-        throw new ValidationAbortedError()
-      }
-
       // eslint-disable-next-line no-await-in-loop
       const result = await rule(value, name, {
         source,
         signal,
         previousValue,
       })
 
-      if (result === undefined) {
-        return
+      if (signal.aborted) {
+        throw new ValidationAbortedError()
+      }
```

Comment:
Moved this to after the rule execution so that we can abort after each rule. Otherwise if validation is aborted during an async rule the rule will still return an error.